pyqtgraph/namedBrush.py

```diff
@@ -0,0 +1,82 @@
+from .Qt import QtGui, QtCore
+from .namedColorManager import NamedColorManager
+
+__all__ = ['NamedBrush']
+DEBUG = False
+
+class NamedBrush(QtGui.QBrush):
+    """ Extends QPen to retain a functional color description """
+    def __init__(self, name, manager=None, alpha=None ):
+        """
+        Creates a new NamedBrush object.
+        'name'    should be in 'functions.Colors'
+        'manager' is a reference to the controlling NamedColorManager
+        'alpha'   controls opacity which persists over palette changes
+        """
+        if DEBUG: print('  NamedBrush created as',name,alpha)
+        super().__init__(QtCore.Qt.SolidPattern) # Initialize QBrush superclass
+        self._identifier = (name, alpha)
+        if manager is None or not isinstance(manager, NamedColorManager):
+            raise ValueError("NamedPen requires NamedColorManager to be provided in 'manager' argument!")
+        self._manager = manager
+        self._updateQColor(self._identifier)
+        self._manager.register( self ) # manually register for callbacks
+
+    def __eq__(self, other): # make this a hashable object
+        # return other is self
+        if isinstance(other, self.__class__):
+            return self._identifier == other._identifier
+        else:
+            return False
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+        
+    def __hash__(self):
+        return id(self)
```

Comment:
We got burned _really_ bad by this and had to fix that here: https://github.com/pyqtgraph/pyqtgraph/pull/1564

At the end of the day, QBrushs/QPens are mutable, and implementing `__hash__` in such a way is our attempt to make them not mutable, which I think we're going to run into issues in general if we try and enforce that criteria.

Reply:
This is only implemented here to allow the NamedColorManager to keep a list of weak references to all defined NamedColor objects. When a NamedColor object is deleted, the reference should be purged from he list immediately. A new object created with the same id would not be part of the list (or set). 
But as discussed on Slack, this logic might be more straightforward to implement as a WeakValueDictionary, with id as the key. That should avoid providing a weak, limited-use hash function here.